Minesweeper/Minesweeper/Entities/MineField.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  MineField.swift
+//  Minesweeper
+//
+//  Created by Dorian Grolaux on 19/02/2018.
+//
+
+import Foundation
+
+typealias Coordinates = (x: Int, y: Int)
+
+class MineField {
+    private let cells: [[Cell]]
+    
+    
+    subscript(x: Int, y: Int) -> Cell? {
+        guard let firstRow = cells.first, firstRow.first != nil else { return nil }
+        
+        let totalRows = cells.count
+        let totalColumns = firstRow.count
+        
+        guard x < totalColumns && y < totalRows else { return nil }
+        return cells[y][x]
+    }
+    
+    /**
+     Initialise the Minefield with an already filled field (use for saved games for instance).
+     - parameter cells: The cell 2d array.
+     **/
+    init(cells: [[Cell]]) {
+        self.cells = cells
+    }
+    
+    
+    /**
+     Initialise the Minefield of given square size and randomly placing the given amount of bombs.
+     - parameter size: The length of each side of the field measured in amount of cells. (Total number of cells is amount squared).
+     - parameter bombs: The amount of bombs to randomly place around in the Minefield.
+     **/
+    init(gridSize size: Int, withBombAmount bombs: Int) {
```

Comment:
We're clearly creating an instance here yet we're calling static functions that will be applied to that instance, why?

Reply:
Somehow you can't call class instance functions during the initializer if they reference `self` at any point. Which is very annoying and why I've made everything static... It's annoying as I wanted the initializer to be the sole thing to call before you could use the instance as you wished, but it's not possible. The other way to remove those pesky statics would be to call the initializer then call a setup function immediately after. 
But then we have to always call 2 functions when we could just call one, and there's also a problem with making the [[Cell]] a `let` which won't be possible anymore... Dilemma